python/src/scipp/compat/mantid.py

```diff
@@ -934,43 +934,54 @@ def to_workspace_2d(x, y, e, coord_dim, instrument_file=None):
+                                         variances=stddevs * stddevs),
+                        coords={dim: coord})
+
+
+def _fit_workspace(ws, mantid_args):
     """
     Performs a fit on the workspace.
 
     :param ws: The workspace on which the fit will be performed
-    :param function: The function used for the fit. This is anything
-                     that mantid.Fit's Function parameter can handle.
-    :param workspace_index: Workspace index which will be fitted.
-    :param start_x: Start X for the fit
-    :param end_x: End X for the fit
     :returns: Dataset containing all of Fit's outputs
     """
     with run_mantid_alg('Fit',
-                        Function=function,
                         InputWorkspace=ws,
-                        WorkspaceIndex=int(workspace_index),
-                        StartX=start_x,
-                        EndX=end_x,
+                        **mantid_args,
                         CreateOutput=True) as fit:
-        ds = sc.Dataset(data={
-            'workspace':
-            sc.Variable(convert_Workspace2D_to_data_array(
-                fit.OutputWorkspace)),
-            'parameters':
-            sc.Variable(convert_TableWorkspace_to_dataset(
-                fit.OutputParameters)),
-            'normalised_covariance_matrix':
-            sc.Variable(
-                convert_TableWorkspace_to_dataset(
-                    fit.OutputNormalisedCovarianceMatrix)),
-        },
-                        attrs={
-                            'status': sc.Variable(fit.OutputStatus),
-                            'chi2_over_DoF':
-                            sc.Variable(fit.OutputChi2overDoF),
-                            'function': sc.Variable(str(fit.Function)),
-                            'cost_function': sc.Variable(fit.CostFunction)
-                        })
-
-        return ds
+        # This is assuming that all parameters are dimensionless. If this is
+        # not the case we should use a dataset with a scalar variable per
+        # parameter instead. Or better, a dict of scalar variables?
+        parameters = convert_TableWorkspace_to_dataset(fit.OutputParameters)
+        parameters = _table_to_data_array(parameters,
+                                          key='Name',
+                                          value='Value',
+                                          stddev='Error')
+        out = convert_Workspace2D_to_data_array(fit.OutputWorkspace)
+        data = sc.Dataset()
+        data['data'] = out['empty', 0]
+        data['calculated'] = out['empty', 1]
+        data['diff'] = out['empty', 2]
+        parameters.attrs['status'] = sc.Variable(fit.OutputStatus)
+        parameters.attrs['chi^2/d.o.f.'] = sc.Variable(fit.OutputChi2overDoF)
+        parameters.attrs['function'] = sc.Variable(str(fit.Function))
+        parameters.attrs['cost-function'] = sc.Variable(fit.CostFunction)
+        return parameters, data
+
+
+def fit(data, mantid_args):
```

Comment:
With changes, almost feels like `fit` doesn't belong in `mantid.compat` - i.e. aside from `mantid_args` user is not aware of dealing with mantid data structures.

Reply:
On the other hand it is far from being usable without Mantid. Same applies to `load`.